catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/transformer/CalmTransformerFeatureTest.scala

```diff
@@ -1,80 +1,75 @@
 package uk.ac.wellcome.transformer
 
+import org.scalatest.concurrent.Eventually
 import org.scalatest.{FunSpec, Matchers}
 import uk.ac.wellcome.models.transformable.CalmTransformable
 import uk.ac.wellcome.models.{
   IdentifierSchemes,
   SourceIdentifier,
   UnidentifiedWork
 }
-import uk.ac.wellcome.test.utils.MessageInfo
-import uk.ac.wellcome.transformer.utils.{
-  TransformableSQSMessageUtils,
-  TransformerFeatureTest
-}
+import uk.ac.wellcome.test.fixtures.{MessageInfo, S3, SnsFixtures, SqsFixtures}
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.transformer.utils.TransformableMessageUtils
 import uk.ac.wellcome.utils.JsonUtil
 import uk.ac.wellcome.utils.JsonUtil._
 
 class CalmTransformerFeatureTest
     extends FunSpec
-    with TransformerFeatureTest
     with Matchers
-    with TransformableSQSMessageUtils {
-
-  override lazy val bucketName: String =
-    "test-calm-transformer-feature-test-bucket"
-  val queueUrl: String = createQueueAndReturnUrl("test_calm_transformer")
-  override val flags: Map[String, String] = Map(
-    "aws.region" -> "eu-west-1",
-    "aws.sqs.queue.url" -> queueUrl,
-    "aws.sqs.waitTime" -> "1",
-    "aws.sns.topic.arn" -> idMinterTopicArn,
-    "aws.metrics.namespace" -> "calm-transformer",
-    "aws.s3.bucketName" -> bucketName
-  )
+    with SqsFixtures
+    with SnsFixtures
+    with S3
+    with fixtures.Server
+    with Eventually
+    with ExtendedPatience
+    with TransformableMessageUtils {
 
-  it(
-    "should poll the dynamo stream for calm data, transform it into unified items and push them into the id_minter SNS topic") {
-    val calmTransformable =
-      CalmTransformable(
-        sourceId = "RecordID1",
-        RecordType = "Collection",
-        AltRefNo = "AltRefNo1",
-        RefNo = "RefNo1",
-        data = """{"AccessStatus": ["public"]}""")
-    val calmHybridRecordMessage =
-      hybridRecordSqsMessage(JsonUtil.toJson(calmTransformable).get, "calm")
-    sqsClient.sendMessage(
-      queueUrl,
-      JsonUtil.toJson(calmHybridRecordMessage).get)
+  it("transforms miro records and publishes the result to the given topic") {
+    withLocalSnsTopic { topicArn =>
+      withLocalSqsQueue { queueUrl =>
+        withLocalS3Bucket { bucketName =>
+          val flags: Map[String, String] = Map(
+            "aws.sqs.queue.url" -> queueUrl,
+            "aws.sns.topic.arn" -> topicArn,
+            "aws.s3.bucketName" -> bucketName,
+            "aws.sqs.waitTime" -> "1",
+            "aws.metrics.namespace" -> "sierra-transformer"
+          ) ++ s3LocalFlags ++ snsLocalFlags ++ sqsLocalFlags
```

Comment:
Q: should we perhaps make these functions that take the bucket name, topic ARN, queue URL, etc., and adds those too?

So the type system forces you to add these flags.

Reply:
Now implemented as #1670.